### PR TITLE
Fix heatmapper tests

### PR DESCRIPTION
**Pull request by fidelram**
Code clean-up. Doctests were moved to unit tests and the tests were updated. Some PEP8 renamings were also done in the process.

### CODE REVIEW

File: deeptools/plotHeatmap.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def plotMatrix(hm, outFileName,
 
     if zMax is None:
         if matrix_flatten is None:
-            matrix_flatten = flattenMatrix(hm.matrix.matrix)
+            matrix_flatten = hm.matrix.matrix.flatten()
```
Comment (bgruening) [inline deeptools/plotHeatmap.py]: This looks suspicious to me.

Comment (fidelram) [inline deeptools/plotHeatmap.py]: the idea is to flatten the matrix only one. thus, be default is set to None. I replaced this by `if not matrix_flatten`

Comment (dpryan79) [inline deeptools/plotHeatmap.py]: I think the question was whether you instead meant `matrix_flatten = hm.matrix.flatten()`.

Comment (bgruening) [inline deeptools/plotHeatmap.py]: Yes, what @dpryan79 was saying. Sorry, I will be more specific next time.

Comment (fidelram) [inline deeptools/plotHeatmap.py]: I see. I changed that in the new PR.

**Pull request merged**